src/reducer/references/possibleValues.js

```diff
@@ -14,13 +14,10 @@ export default (previousState = initialState, { type, payload, meta }) => {
     }
 };
 
-export const getPossibleReferences = (state, referenceSource, reference, selectedId) => {
-    if (!state.admin.references.possibleValues[referenceSource]) {
-        return typeof selectedId === 'undefined' || !state.admin[reference].data[selectedId] ? [] : [state.admin[reference].data[selectedId]];
-    }
-    const possibleValues = state.admin.references.possibleValues[referenceSource];
-    if (typeof selectedId !== 'undefined' && !possibleValues.includes(selectedId)) {
-        possibleValues.unshift(selectedId);
+export const getPossibleReferences = (state, referenceSource, reference, selectedIds = []) => {
+    const possibleValues = state.admin.references.possibleValues[referenceSource] || [];
+    if (selectedIds.length !== 0) {
+        selectedIds.forEach(id => possibleValues.includes(id) || possibleValues.unshift(id));
```

Comment:
Aren't you mutating the current state here ? It would probably be safer to initialize `possibleValues` from state with `Array.from`